numpy/_core/tests/test_stringdtype.py

```diff
@@ -740,6 +740,12 @@ def test_add_promoter(string_list):
         assert_array_equal(arr + op, rresult)
 
 
+def test_add_promoter_reduce():
+    # Exact TypeError could change, but ensure StringDtype doesn't match
+    with pytest.raises(TypeError, match="the resolved dtypes are not"):
+        np.add.reduce(np.array(["a", "b"], dtype="U"))
```

Comment:
Might it be worth testing too that it does work if one passes `dtype="T"` to the reduction?